mls-rs-core/src/lib.rs

```diff
@@ -24,3 +24,42 @@ pub use mls_rs_codec;
 
 #[cfg(feature = "arbitrary")]
 pub use arbitrary;
+
+#[cfg(feature = "serde")]
+pub mod zeroizing_serde {
+    use alloc::vec::Vec;
+    use serde::{Deserializer, Serializer};
+    use zeroize::Zeroizing;
+
+    use crate::vec_serde;
+
+    pub fn serialize<S: Serializer>(v: &Zeroizing<Vec<u8>>, s: S) -> Result<S::Ok, S::Error> {
+        vec_serde::serialize(v, s)
+    }
+
+    pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<Zeroizing<Vec<u8>>, D::Error> {
+        vec_serde::deserialize(d).map(Zeroizing::new)
+    }
+}
+
+#[cfg(feature = "serde")]
+pub mod vec_serde {
+    use alloc::{string::String, vec::Vec};
+    use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+    pub fn serialize<S: Serializer>(v: &Vec<u8>, s: S) -> Result<S::Ok, S::Error> {
+        if s.is_human_readable() {
+            String::serialize(&hex::encode(v), s)
+        } else {
+            Vec::<u8>::serialize(&v, s)
+        }
+    }
+
+    pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<Vec<u8>, D::Error> {
+        if d.is_human_readable() {
+            hex::decode(String::deserialize(d)?).map_err(Error::custom)
+        } else {
+            Vec::<u8>::deserialize(d)
```

Comment:
This should use the `serde_bytes` crate for efficient and flexible serialization of bytes and do this:
```rust
serde_bytes::deserialize(d)
```

`serde_bytes` is a common dependency when using `serde`, and it does not bring any additional dependency.